# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/fget/hotfix-gcc14.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- fget/fget.c.vanilla	2024-03-28 20:19:39.893209999 +0100
+++ fget/fget.c	2024-03-28 20:21:00.382205945 +0100
@@ -1,6 +1,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <sys/types.h>
+
 #ifndef NODNS
 #include "dns.h"
 #endif
@@ -150,7 +151,7 @@
   stralloc_cats(request,"\r\n\r\n");
 }
 
-int isdigit(char c) {
+int isdigit(int c) {
   return (c>='0' && c<='9');
 }
 
--- fget/buffer.h.vanilla	2024-03-28 20:21:20.811204916 +0100
+++ fget/buffer.h	2024-03-28 20:32:31.512171132 +0100
@@ -6,10 +6,10 @@
   unsigned int p;
   unsigned int n;
   int fd;
-  int (*op)();
+  long (*op)(int, const void *, unsigned long);
 } buffer;
 
-#define BUFFER_INIT(op,fd,buf,len) { (buf), 0, (len), (fd), (op) }
+#define BUFFER_INIT(op,fd,buf,len) { (buf), 0, (unsigned int)(len), (fd), (op) }
 #define BUFFER_INSIZE 8192
 #define BUFFER_OUTSIZE 8192
 
--- fget/buffer_put.c.vanilla	2024-03-28 20:32:11.336172148 +0100
+++ fget/buffer_put.c	2024-03-28 20:32:57.046169845 +0100
@@ -3,7 +3,8 @@
 #include "byte.h"
 #include "error.h"
 
-static int allwrite(int (*op)(),int fd,const char *buf,unsigned int len)
+static int allwrite(long (*op)(int, const void *, unsigned long),
+	int fd,const char *buf,unsigned int len)
 {
   int w;
 
